modules/nixidy.nix

```
@@ -220,6 +220,11 @@ in {
                     // (lib.optionalAttrs (lib.length app.syncPolicy.finalSyncOpts > 0) {
                       syncOptions = app.syncPolicy.finalSyncOpts;
                     });
+
+                  ignoreDifferences =
+                    if builtins.isAttrs app.ignoreDifferences
+                    then lib.mapAttrsToList (kind: value: value // {inherit kind;}) app.ignoreDifferences
+                    else null;
```

Turns out the code generated by `fromCRD` for ArgoCD Applications is already expecting an attrset and converts it to a list for us. Therefore we can just pass it on directly.

See: https://github.com/arnarg/nixidy/blob/d1de45d614b9ef97c250caecd93516932c66579d/modules/generated/argocd.nix#L2042

```suggestion
                  ignoreDifferences = app.ignoreDifferences
```

But it'd be better to just add it to the inherit in the top of `spec` (where there's `inherit (app) project` and change it to `inherit (app) project ignoreDifferences`).